### PR TITLE
[QOL-5815] launch additional test runs for Publications infrastructure

### PR DESCRIPTION
**Pull request by ThrawnCA**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -31,23 +31,43 @@ aliases:
...
       DEPLOY_ENV: STAGING
     steps: *steps_config
 
 workflows:
   version: 2
   main:
     jobs:
-      - build
-      - build_staging
+      - OpenData_DEV
```
Comment (duttonw) [inline .circleci/config.yml:70]: look at having dev's run first then trigger staging
https://circleci.com/docs/2.0/configuration-reference/#requires

look at adding cron to trigger once a month at min
https://circleci.com/docs/2.0/configuration-reference/#cron

**Review by duttonw — APPROVED** [review]

**Comment by duttonw** [comment]
pub crashed. nice.

might also need to look at having 2 different set's for selenium tests as this would be the best place for integration testing ;)

File: .circleci/config.yml (hunk 1 above)
Comment (ThrawnCA) [reply .circleci/config.yml:70]: Having staging depend on dev could save some unnecessary runs in cases of failure, but would double the length of the build...is it worthwhile?

A cron trigger should be simple enough, though.

Comment (duttonw) [reply .circleci/config.yml:77]: circleci Free
1 concurrent job with access to Linux and Windows builds on Small and Medium sizes
we only have 7% left for the next 3 days.
doing dev first should be helpful but this repo might be better off being migrated to github actions to remove the credit limit.

**Review by duttonw — APPROVED** [review]

**Comment by ThrawnCA** [comment]
> pub crashed. nice.

I think it's a result of how we're reusing `ckanext-data-qld` but renaming it. May need to put in some special handling.

**Comment by ThrawnCA** [comment]
...And CircleCI has stopped building it. Maybe let's go with Github Actions.

(Is there any point in making Staging dependent on Dev when it's all free?)

File: .docker/test-Publications.ini [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+
+## Plugins Settings
+
+# Note: Add ``datastore`` to enable the CKAN DataStore
+#       Add ``datapusher`` to enable DataPusher
+#       Add ``resource_proxy`` to enable resorce proxying and get around the
+#       same origin policy
+# @todo:setup Cleanup the list to use only required plugins.
+ckan.plugins = stats text_view image_view recline_view data_qld_resources publications_qld_theme qgovext
```
Comment (duttonw) [inline .docker/test-Publications.ini:97]: is this semi dynamic on our real life website?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        run: flake8 . --count --statistics
+
+  test:
+    needs: lint
+    strategy:
+      matrix:
+        # Not ready for 2.9 yet
+        # ckan-version: [2.9, 2.9-py2, 2.8]
+        ckan-version: [2.8]
+        application: [OpenData, Publications]
```
Comment (duttonw) [inline .github/workflows/test.yml:36]: nice

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      VARS_TYPE: ${{ matrix.application }}
+      DEPLOY_ENV: ${{ matrix.target_environment }}
+
+    steps:
+      - name: Update CKAN version
+        run: |
+          cd /srv/app/src/ckan
+          git remote set-url origin https://github.com/qld-gov-au/ckan.git
+          # Extracting the version would be better, but would require
+          # Jinja rendering as well as YAML parsing
```
Comment (duttonw) [inline .github/workflows/test.yml:67]: might be able to do it in a git actions container to setup variables to be used see https://github.com/qld-gov-au/gha-publish-to-git which is used to publish to git or https://github.com/qld-gov-au/foodpantry-kyfb/blob/master/.github/workflows/publish.yml#L53 on setting variable from a bash script to be injected later (ie could run a python script etc)

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      - name: Setup extension (CKAN >= 2.9)
+        if: ${{ matrix.ckan-version != '2.7' && matrix.ckan-version != '2.8' }}
+        run: |
+          ckan -c test-$VARS_TYPE.ini db init
+          ckan -c test-$VARS_TYPE.ini datastore set-permissions | sudo -u postgres psql -p $PG_PORT
+      - name: Setup extension (CKAN < 2.9)
+        if: ${{ matrix.ckan-version == '2.7' || matrix.ckan-version == '2.8' }}
+        run: |
+          paster --plugin=ckan db init -c test-$VARS_TYPE.ini
+          paster datastore set-permissions -c test-$VARS_TYPE.ini | sudo -u postgres psql -p $PG_PORT
```
Comment (duttonw) [inline .github/workflows/test.yml:90]: what about the other plugins which might need to be inited ?

**Review by duttonw — APPROVED** [review]

File: .docker/test-Publications.ini (hunk 2 above)
Comment (ThrawnCA) [reply .docker/test-Publications.ini:97]: Yes, it's assembled by the OpsWorks cookbook.

File: .github/workflows/test.yml (hunk 5 above)
Comment (ThrawnCA) [reply .github/workflows/test.yml:90]: Future enhancement after we can get this to run?

**Review by duttonw — APPROVED** [review]

**Comment by ThrawnCA** [comment]
Obsolete, see https://github.com/qld-gov-au/ckan-qld-infrastructure/pull/118

**Pull request closed without merging**